src/server/routes/search.rs

```diff
@@ -236,24 +228,3 @@ fn is_match_from_filter_list(
 
     Ok(false)
 }
-
-/// A helper function which returns the safe search level based on the url params
-/// and cookie value.
-///
-/// # Argurments
-///
-/// * `safe_search` - Safe search level from the url.
-/// * `cookie` - User's cookie
-/// * `default` - Safe search level to fall back to
-fn get_safesearch_level(safe_search: &Option<u8>, cookie: &Option<u8>, default: u8) -> u8 {
-    match safe_search {
-        Some(ss) => {
-            if *ss >= 3 {
-                default
-            } else {
-                *ss
-            }
-        }
-        None => cookie.unwrap_or(default),
-    }
-}
```

Comment:
I would suggest keeping this function and using it in the code because this function basically takes the safe_search_level values from the config, search URL parameter and the cookie. If the config safe_search_level is 3 or 4 then it uses the config's safe_search_level. This disallows the user from overriding it by the cookie or the search URL parameters. Also, this function also, checks whether the search URL parameter was provided if not then it falls back to using the cookie's safe_search_level (if cookie was set) otherwise it falls back to using the config's safe_search_level. 

The reason behind having this safe_search_level handling function is to allow the user (server administrator) to provide an SFW only server where the search results do not contain any NSFW search results (content). I hope you got the idea behind having this function. :slightly_smiling_face: .

Reply:
Yeah I thought my code already handled all the cases but I forgot about the safe search level from the params in the URL. But I noticed we were now handling the potential cookie and then config safe search levels twice. 

We can use a single line like this `if let Some(level) = &params.safesearch { search_settings.safe_search_level = *level; }` or a match statement if preferred. `match &params.safesearch {
                Some(level) => { search_settings.safe_search_level = *level; },
                None => {},
            }`

and then have the same result

Reply:
Could you explain why levels 3 and above fall back to the default in the URL? Is it because those levels aren't implemented correctly in some search engines?

Reply:
> Could you explain why levels 3 and above fall back to the default in the URL? Is it because those levels aren't implemented correctly in some search engines?
> 
> Edit: To keep checking for that, we can modify get_safesearch_level itself to work like this
> 
> `search_settings.safe_search_level = get_safesearch_level(&params.safesearch, config.safe_search);`
> 
> ```
> fn get_safesearch_level(safe_search: &Option<u8>, default: u8) -> u8 {
>     match safe_search {
>         Some(ss) => {
>             if *ss >= 3 {
>                 default
>             } else {
>                 *ss
>             }
>         }
>         None => default,
>     }
> }
> ```

No, actually if notice in search engines they fall to level 2 this is because in search engines the maximum `safe search level`  tends to be 2 and they are mean't to remove NSFW results but you see the sad truth is they most of the time don't remove NSFW results properly so to completely eliminate those results we have implemented a filter list method which allows the user to define custom filters (which is regex based) to remove explicit NSFW completely which gives more control to the user of what results they want to see/provide. I hope you got my point :slightly_smiling_face: .

Reply:
I see, so they would need to modify the config file and set up their filter list as wanted to do that. Is there somewhere else in the code that enforces the safe search level to be that high?

Reply:
> I see, so they would need to modify the config file and set up their filter list as wanted to do that. Is there somewhere else in the code that enforces the safe search level to be that high?

Sorry for the late reply.

Yes, there are checks in the `aggregator.rs` file where it checks for the safe search level is equal to 3 or not and filters the results if the condition is true otherwise returns the search results as it is. :slightly_smiling_face: